### ISSUE TITLE
Cannot send email

### ISSUE THREAD

**Issue by ramihamati** [sendinblue/APIv3-csharp-library#18]
I am trying to use this package, I have an account set up. I managed to add a contact using the example but when I try to send an email I have the following error
```
  Configuration.Default.ApiKey.Add("api-key", "IREMOVEDKEY");

            var apiInstance = new AccountApi();

            var smtpApi = new SMTPApi();

            var email = new SendSmtpEmail(new SendSmtpEmailSender
            {
                Email = "test@mydomain.com",
                Name = "me",
            }, new System.Collections.Generic.List<SendSmtpEmailTo>
                {
                    new SendSmtpEmailTo
                    {
                        Email = "useremail@gmail.com",
                        Name = "Yo Man!"
                    }
                }, null, null, null, "Hello", "Hello");

            smtpApi.SendTransacEmail(email);
```

System.IO.InvalidDataException: 'email is a required property for SendSmtpEmailSender and cannot be null'

**Comment by amitsendinblue** [comment]
Please try below code:

```
Configuration.Default.ApiKey.Add("api-key", "YOUR-API-KEY");
var apiInstance = new SMTPApi();
string SenderName = "amit";
string SenderEmail = "amit.joshi@sendinblue.com";
SendSmtpEmailSender Email = new SendSmtpEmailSender(SenderName, SenderEmail);
string ToEmail = "amit.joshi@sendinblue.com";
string ToName = "Amit";
SendSmtpEmailTo smtpEmailTo = new SendSmtpEmailTo(ToEmail, ToName);
List<SendSmtpEmailTo> To = new List<SendSmtpEmailTo>();
To.Add(smtpEmailTo);
string BccName = "Amit";
string BccEmail = "amit.joshi+1@sendinblue.com";
SendSmtpEmailBcc BccData = new SendSmtpEmailBcc(BccEmail, BccName);
List<SendSmtpEmailBcc> Bcc = new List<SendSmtpEmailBcc>();
Bcc.Add(BccData);
string CcName = "Amit";
string CcEmail = "ektagupta@sendinblue.com";
SendSmtpEmailCc CcData = new SendSmtpEmailCc(CcEmail, CcName);
List<SendSmtpEmailCc> Cc = new List<SendSmtpEmailCc>();
Cc.Add(CcData);
string HtmlContent = "<h1>Hello this is test</h1>";
string TextContent = "Hello this is test";
string Subject = "Test Subject";
string ReplyToName = "Amit";
string ReplyToEmail = "amit.joshi@sendinblue.com";
SendSmtpEmailReplyTo ReplyTo = new SendSmtpEmailReplyTo(ReplyToEmail, ReplyToName);
string AttachmentUrl = null;
string stringInBase64 = "SGVsbG8gdGhpcyBpcyB0ZXN0";
byte[] Content = System.Convert.FromBase64String(stringInBase64);
string AttachmentName = "test.txt";
SendSmtpEmailAttachment AttachmentContent = new SendSmtpEmailAttachment(AttachmentUrl, Content, AttachmentName);
List<SendSmtpEmailAttachment> Attachment = new List<SendSmtpEmailAttachment>();
Attachment.Add(AttachmentContent);
JObject Headers = new JObject();
Headers.Add("x-mailin-custom", "C#Test");
long? TemplateId = null;
JObject Params = null;
List<string> Tags = new List<string>();
Tags.Add("C# Wrapper");
try
{
var sendSmtpEmail = new SendSmtpEmail(Email, To, Bcc, Cc, HtmlContent, TextContent, Subject, ReplyTo, Attachment, Headers, TemplateId, Params, Tags);
CreateSmtpEmail result = apiInstance.SendTransacEmail(sendSmtpEmail);
Console.WriteLine(result.ToJson());
Console.ReadLine();
}
catch (Exception e)
{
Console.WriteLine(e.Message);
Console.ReadLine();
}
```

**Comment by amitsendinblue** [comment]
I hope this solution will work fine, closing the issue.

**Comment by ramihamati** [comment]
Thanks for the solution. I will try it.

**Comment by ramihamati** [comment]
Sorry for the late response. I did followed on your example, and instead of using an object initializer I passed the parameters in the constructor. 

This solution works, although I find it confusing that you allow for null values in the constructor and immediately validate them. Object initializer will first call the ctor and then initialize the properties. I don't really see the use of allowing null values in the constructor, as long as the models have the [JsonConstructor] on a parameterless different ctor.
